Source/Events.Filters/Internal/EventFilterProcessor.cs

```diff
@@ -0,0 +1,114 @@
+            FilterRegistrationRequest registerArguments,
+            Func<FilterEventRequest, CancellationToken, Task<FilterResponse>> callback,
+            uint pingTimeout,
+            CancellationToken cancellation)
+            => _reverseCallClientsCreator.Create(
+                RegisterArguments,
+                this,
+                new DuplexStreamingMethodCaller(),
+                new ReverseCallMessageConverter());
+
+        /// <inheritdoc/>
+        protected override FilterResponse CreateResponseFromFailure(ProcessorFailure failure)
+            => new FilterResponse { Failure = failure };
+
+        /// <inheritdoc/>
+        protected override async Task<FilterResponse> Filter(object @event, EventContext context)
+            => new FilterResponse { IsIncluded = await _filterEventCallback(@event, context).ConfigureAwait(false) };
+
+        class DuplexStreamingMethodCaller : ICanCallADuplexStreamingMethod<FiltersClient, FilterClientToRuntimeMessage, FilterRuntimeToClientMessage>
+        {
+            public AsyncDuplexStreamingCall<FilterClientToRuntimeMessage, FilterRuntimeToClientMessage> Call(Channel channel, CallOptions callOptions)
+                => new FiltersClient(channel).Connect(callOptions);
+        }
+
+        class ReverseCallMessageConverter : IConvertReverseCallMessages<FilterClientToRuntimeMessage, FilterRuntimeToClientMessage, FilterRegistrationRequest, FilterRegistrationResponse, FilterEventRequest, FilterResponse>
+        {
+            public FilterClientToRuntimeMessage CreateMessageFrom(FilterRegistrationRequest arguments)
+                => new FilterClientToRuntimeMessage { RegistrationRequest = arguments };
+
+            public FilterClientToRuntimeMessage CreateMessageFrom(Pong pong)
+                => new FilterClientToRuntimeMessage { Pong = pong };
+
+            public FilterClientToRuntimeMessage CreateMessageFrom(FilterResponse response)
+                => new FilterClientToRuntimeMessage { FilterResult = response };
+
+            public FilterRegistrationResponse GetConnectResponseFrom(FilterRuntimeToClientMessage message)
+                => message.RegistrationResponse;
+
+            public Ping GetPingFrom(FilterRuntimeToClientMessage message)
+                => message.Ping;
+
+            public ReverseCallRequestContext GetRequestContextFrom(FilterEventRequest message)
+                => message.CallContext;
+
+            public FilterEventRequest GetRequestFrom(FilterRuntimeToClientMessage message)
+                => message.FilterRequest;
+
+            public void SetConnectArgumentsContextIn(ReverseCallArgumentsContext context, FilterRegistrationRequest arguments)
+                => arguments.CallContext = context;
+
+            public void SetResponseContextIn(ReverseCallResponseContext context, FilterResponse response)
+                => response.CallContext = context;
+        }
```

Comment:
I think that these implementations are the exceptions to the rule. Since these implementations has no other purpose what so ever other than here. I would prefer not having these in their own files, but I don't know if I'm the only one thinking that? Thoughts, @jakhog @einari?